helix-core/src/main/java/org/apache/helix/task/JobConfig.java

```diff
@@ -553,7 +556,11 @@ public static Builder fromMap(Map<String, String> cfg) {
       }
       if (cfg.containsKey(JobConfigProperty.RebalanceRunningTask.name())) {
         b.setRebalanceRunningTask(
-            Boolean.valueOf(cfg.get(JobConfigProperty.RebalanceRunningTask.name())));
+            Boolean.parseBoolean(cfg.get(JobConfigProperty.RebalanceRunningTask.name())));
+      }
+      if (cfg.containsKey(ZNRecord.ENABLE_COMPRESSION_BOOLEAN_FIELD)) {
```

Comment:
It is redundant to check `containsKey`: if the value is not set, parseBoolean still returns false.
Unless the default value enableCompression is true, but I don't think this should be the case.